### PR TITLE
SQL: Taking in account Daylight Saving Time for GROUP BY PERIOD #4131

### PR DESCRIPTION
**Pull request by vtols**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            "SELECT count(*), " +
+            "date_format(time, 'yyyy-MM-dd HH:mm', '%1$s'), " +
+            "date_format(time, 'yyyy-MM-dd HH:mm', 'UTC') " +
+            "FROM '%2$s' " +
+            "GROUP BY PERIOD(1 DAY, '%1$s')";
+
+    // three days
+    private static final int hours = 24 * 3;
+    private static final long hourDuration = 60 * 60 * 1000;
+
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java]: Please, use upper case for constant values.

File: src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    private static final String QUERY_TEMPLATE =
+            "SELECT count(*), " +
+            "date_format(time, 'yyyy-MM-dd HH:mm', '%1$s'), " +
+            "date_format(time, 'yyyy-MM-dd HH:mm', 'UTC') " +
+            "FROM '%2$s' " +
+            "GROUP BY PERIOD(1 DAY, '%1$s')";
+
+    // three days
+    private static final int HOURS = 24 * 3;
+    private static final long HOUR_DURATION = 60 * 60 * 1000;
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java]: Let's move it to some class with constants.

File: src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    private static final String METRIC_NAME3 = TestNames.metric();
+    private static final String QUERY_TEMPLATE =
+            "SELECT count(*), " +
+            "date_format(time, 'yyyy-MM-dd HH:mm', '%1$s'), " +
+            "date_format(time, 'yyyy-MM-dd HH:mm', 'UTC') " +
+            "FROM '%2$s' " +
+            "GROUP BY PERIOD(1 DAY, '%1$s')";
+
+    // three days
+    private static final int HOURS = 24 * 3;
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java]: Why have you chosen such name? Do we really need this constant?

File: src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java (hunk 2 above)
Comment (vtols) [inline src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java]: They aren't used anywhere else, is it really a good idea?

File: src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java (hunk 3 above)
Comment (vtols) [inline src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java]: I can just inline them if you want

File: src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java (hunk 2 above)
Comment (vtols) [inline src/test/java/com/axibase/tsd/api/method/sql/function/period/align/PeriodDaylightSavingTimeTest.java]: Will update this PR after merging https://github.com/axibase/atsd-api-test/pull/312

**Pull request merged**